### PR TITLE
Creation info on backend

### PR DESCRIPTION
**Pull request by thenakedthunder**

### CODE REVIEW

File: flack/backend/channel.py [hunk 1]
```diff
@@ -4,22 +4,75 @@
 properties
 """
 
-from datetime import datetime
+from datetime import datetime, timedelta
```
Comment (thenakedthunder) [inline flack/backend/channel.py:7]: timedelta not used, delete it

File: flack/backend/channel.py [hunk 2]
```diff
@@ -4,22 +4,75 @@
...
 import message
+import enum
```
Comment (thenakedthunder) [inline flack/backend/channel.py:9]: to be deleted

File: flack/backend/channel.py [hunk 3]
```diff
@@ -4,22 +4,75 @@
...
+            return "today"
+        elif diff_in_days == 1:
+            return "yesterday"
+        elif diff_in_days < 7:
+            return self.creation_time.strftime("on %A")     #returns day name
+        else:
+            return self._get_formatted_date()       #returns full date
+
+    def _get_time_of_creation(self):
+        """WARNING: '%#H' works on Windows, but on UNIX??? ('%-H' can be tried
```
Comment (thenakedthunder) [inline flack/backend/channel.py:55]: is it possible to write some check for the system type?

File: flack/backend/channel.py [hunk 4]
```diff
@@ -4,22 +4,75 @@
...
+        """an error is thrown when the creation time is later than the current
+        time, as this should not happen"""
+        if self.creation_time > datetime.now():
+            raise ValueError("Invalid creation time: it is later than current system time")
+
+    def _get_how_many_days_ago_was_creation(self):
+        return (datetime.today().date() - self.creation_time.date()).days
+   
+    def _get_formatted_date(self):
+        """WARNING: '%#d' works on Windows, but on UNIX??? ('%-d' can be tried
```
Comment (thenakedthunder) [inline flack/backend/channel.py:71]: is it possible to write some check for the system type?

File: flack/backend/channel_tests.py [hunk 5]
```diff
@@ -11,30 +13,278 @@ class channel_tests(unittest.TestCase):
...
     """Unit Tests for channel.py"""
 
+    """Tests for check_if_creation_time_is_valid"""
+
+    def test__check_if_creation_time_is_valid_with_valid_input(self):
+        """should NOT throw an exception when creation time is earlier than the
+        current system time
+        """
+        test_channel = Channel("Kerekes", "Tanyasi Jóska")
+        test_channel.creation_time = datetime.now()
```
Comment (thenakedthunder) [inline flack/backend/channel_tests.py:28]: same happens in Channel's init, this isn't needed here

File: flack/backend/channel_tests.py [hunk 6]
```diff
@@ -11,30 +13,278 @@ class channel_tests(unittest.TestCase):
...
+        with self.assertRaises(ValueError): 
+            test_channel._check_if_creation_time_is_valid()
+
+
+    """Tests for get_how_many_days_ago_was_creation"""
+    def test_get_how_many_days_ago_was_creation_today(self):
+        """should return 0 if the creation day is the current day"""
+        test_channel = Channel("Kerekes", "Tanyasi Jóska")
+        #input without time details
+        test_channel.creation_time = datetime.today()
```
Comment (thenakedthunder) [inline flack/backend/channel_tests.py:63]: delete this

File: flack/backend/channel_tests.py [hunk 7]
```diff
@@ -11,30 +13,278 @@ class channel_tests(unittest.TestCase):
...
+    def test_get_how_many_days_ago_was_creation_365_days_ago(self):
+        """should return 365 when the current date and the creation date are 
+        365 days apart. 
+        NOTE: using days instead of years is intentional (leap years!)
+        """
+        test_channel = Channel("My Ballad", "John Henry")
+        test_channel.creation_time = datetime.today() - timedelta(365)
+
+        self.assertEqual(test_channel._get_how_many_days_ago_was_creation(),
+                        365)
```
Comment (thenakedthunder) [inline flack/backend/channel_tests.py:100]: make this 366 days

File: flack/backend/channel_tests.py [hunk 8]
```diff
@@ -11,30 +13,278 @@ class channel_tests(unittest.TestCase):
...
+        self.assertEqual(test_channel._get_time_of_creation(), "14:30")
+
+
+    """Tests for get_day_of_creation"""
+
+    def test_get_day_of_creation_on_same_day(self):
+        """should return 'today' if the creation date is the same as the
+        current date"""
+        test_channel = Channel("Cream", "Strange Brew")
+        test_channel.creation_time = datetime.today()
```
Comment (thenakedthunder) [inline flack/backend/channel_tests.py:166]: no need for this

File: flack/backend/channel_tests.py [hunk 9]
```diff
@@ -11,30 +13,278 @@ class channel_tests(unittest.TestCase):
...
+        test_channel.creation_time = datetime.today() - timedelta(6)
+
+        expected_result = test_channel.creation_time.strftime("on %A")
+        self.assertEqual(test_channel._get_day_of_creation(), expected_result)
+        self.assertIn(expected_result.split()[1], self.day_names)
+
+    def test_get_day_of_creation_from_7_days_back(self):
+        """should return the correct format without year if the creation date 
+        was at least a week before the current date
+        WARNING! in the first week of the year this won't test every edge case
```
Comment (thenakedthunder) [inline flack/backend/channel_tests.py:191]: rewrite this warning, not concrete enough

File: flack/src/ChannelListPanel/ChannelListPanel.tsx [hunk 10]
```diff
@@ -67,7 +65,19 @@ export default function ChannelListPanel() {
...
-    // -------------- RENDERING COMPONENTS ---------------
+    // -------------- COMPONENTS ---------------
+
+    const getSecondaryChannelText = (channel: Channel) => {
+        const { messages, creatorDisplayName, creationTime} = channel
+
+        if (messages.length === 0) {
+            return `Created ` + creationTime + ` by ${creatorDisplayName}`
+        }
+
```
Comment (thenakedthunder) [inline flack/src/ChannelListPanel/ChannelListPanel.tsx:76]: maybe for else case: no-op or error throwing?

**Review by thenakedthunder — COMMENTED** [review]
Changes requested.

File: flack/backend/channel.py (hunk 3 above)
Comment (thenakedthunder) [reply flack/backend/channel.py:55]: it works on Raspberry Pi too, so for now this looks OK

File: flack/backend/channel.py (hunk 4 above)
Comment (thenakedthunder) [reply flack/backend/channel.py:71]: On Raspbian it seems to be working
